docs/rate-limiting.md

```diff
@@ -0,0 +1,81 @@
+# Rate Limiting
+
+Rate limiting controls the rate of requests that can be forwarded by a Hyperbahn
+process. The motivation behind is that a service should never stress Hyperbahn in
+such an unexpected way that the quality of other services gets affected.
+In general, rate limiting can be considered as a firewall between services.
+
+
+## Goals
+
+Hyperbahn serves as a communication bus for services. In practice, a service
+can have a surge of  requests due to various of reasons such as sudden increase of
+demands or failures/retries. When such events occur, these requests may saturate the 
+hyperbahn nodes and cause a service quality degradation.
+
+There are two goals of rate limiting:
+
+* Control the RPS (Request Per Second) that a service can have on a hyperbahn node.
+* Control the total RPS that a hyperbahn node should handle.
+
+Non-goals:
+
+* Rate limiting is not used to limit the healthy growth of a service's requests.
+* Rate limiting is not [Circuit Breaking](./circuit-breaking.md). It treats all
+  requests the same.
+* Rate limiting is not SLA (Service Level Agreement). It only guarantees to control
```

Comment:
Rate limiting is not for enforcing SLAs (Service Level Agreements). Plural agreement.